dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -0,0 +1,249 @@
+                host != null && InodeUtils.isSet(host.getInode()) ? host.getIdentifier()
+                        : Host.SYSTEM_HOST, VanityUrlUtil.fixURI(uri), languageId));
+        if (result == null || !InodeUtils.isSet(result.getInode())) {
+            List<VanityUrl> results = new ArrayList<>();
+            String hostId = (host != null ? host.getIdentifier() : Host.SYSTEM_HOST);
+            try {
+                List<Contentlet> contentResults = contentletAPI
+                        .search("+baseType:" + BaseContentType.VANITY_URL.getType()
+                                + " +languageId:" + languageId + " +conHost:" + hostId
+                                + " +vanityUrl:" + VanityUrlUtil.fixURI(uri) + (live ? " +live:true"
+                                : " +working:true"), 0, 0, StringPool.BLANK, user, false);
+                contentResults.stream().forEach((Contentlet con) -> {
+                    VanityUrl vanityUrl = getVanityUrlFromContentlet(con);
+                    try {
+                        if (con.isLive()) {
+                            addToVanityURLCache(vanityUrl);
+                        } else {
+                            VanityUrlServices.invalidateVanityUrl(vanityUrl);
+                        }
+                    } catch (DotDataException | DotSecurityException e) {
+                        Logger.error(this, "Error processing Vanity Url - contentlet Id:" + con
+                                .getIdentifier(), e);
+                    }
+                    results.add(vanityUrl);
+                });
+
+                List<ContentType> vanityUrlContentTypes = APILocator.getContentTypeAPI(user)
+                        .findByType(BaseContentType.VANITY_URL);
+
+                if (results.size() == 0 && vanityUrlContentTypes.size() > 0
+                        && ((VanityUrlContentType) vanityUrlContentTypes.get(0)).fallback()) {
+                    long defaultLanguageId = APILocator.getLanguageAPI().getDefaultLanguage()
+                            .getId();
+                    contentResults = contentletAPI
+                            .search("+baseType:" + BaseContentType.VANITY_URL.getType()
+                                            + " +languageId:" + defaultLanguageId + " +conHost:" + hostId
+                                            + " +vanityUrl:" + VanityUrlUtil.fixURI(uri) + (live
+                                            ? " +live:true" : " +working:true"), 0, 0, StringPool.BLANK,
+                                    user, false);
+                    contentResults.stream().forEach((Contentlet con) -> {
+                        VanityUrl vanityUrl = getVanityUrlFromContentlet(con);
+                        try {
+                            if (con.isLive()) {
+                                addToVanityURLCache(vanityUrl);
+                            } else {
+                                VanityUrlServices.invalidateVanityUrl(vanityUrl);
```

Comment:
We have to keep in mind that using static methods then prevent using a mock for unit testing